scipy/integrate/_ivp/radau.py

```diff
@@ -441,12 +473,16 @@ def _step_impl(self):
             converged = False
             while not converged:
                 if LU_real is None or LU_complex is None:
-                    LU_real = self.lu(MU_REAL / h * self.I - J)
-                    LU_complex = self.lu(MU_COMPLEX / h * self.I - J)
+                    if self.mass_matrix is None:
+                        LU_real = self.lu(MU_REAL / h * self.I - J)
+                        LU_complex = self.lu(MU_COMPLEX / h * self.I - J)
+                    else:
+                        LU_real = self.lu(MU_REAL / h * self.mass_matrix - J)
+                        LU_complex = self.lu(MU_COMPLEX / h * self.mass_matrix - J)
```

Comment:
I have chosen to let the initial implementation in the code, using the if statement to switch to the formulation with a mass matrix if necessary. Another more compact solution would be to simply assume that the mass matrix is the identity if it is not supplied, and only use the mass matrix formulation.
The same can be said regarding the error estimation further down.

Reply:
I have corrected this. If it is not given by the user, the mass matrix is now replaced by a sparse identity matrix (which was already available as `self.I` in the Radau class).